modules/heat_transfer/src/bcs/ADConvectiveHeatFluxBC.C

```diff
@@ -105,10 +105,13 @@ ADConvectiveHeatFluxBC::initialSetup()
 
     // Determine if neighbor can be used, just in case
     const auto neighbor = elem->neighbor_ptr(side);
-    T_inf_can_use_neighbor = T_inf_can_use_neighbor && neighbor &&
-                             _T_infinity_functor->hasBlocks(neighbor->subdomain_id());
-    htc_can_use_neighbor =
-        htc_can_use_neighbor && neighbor && _htc_functor->hasBlocks(neighbor->subdomain_id());
+    if (neighbor && !neighbor->is_remote())
+    {
+      T_inf_can_use_neighbor =
+          T_inf_can_use_neighbor && _T_infinity_functor->hasBlocks(neighbor->subdomain_id());
+      htc_can_use_neighbor =
+          htc_can_use_neighbor && _htc_functor->hasBlocks(neighbor->subdomain_id());
+    }
```

Comment:
This changes the logic. If initially `T_inf_can_use_neighbor` and `!neighbor` your PR will change from a final result of `T_inf_can_use_neighbor == false` to `T_inf_can_use_neighbor == true`. This indicates we don't have test coverage for this if no tests are failing

Reply:
I'm trying to figure out what's going on with the test coverage for this.  The only test that sets `T_infinity_functor` and `heat_transfer_coefficient_functor` is `fe_fv_coupled.i` ... but that test *isn't* restricted to ReplicatedMesh.

And in hindsight, I'm not sure how Yaqi is hitting a `neighbor->is_remote()` case.  This BC should be getting evaluated only on local elements, right?  Those should only have ghosted neighbors, not remote neighbors.

Reply:
@lindsayad yes, this does change the logic. When neighbor is null, `_T_infinity_use_neighbor` need to be true to trigger the error, which means the functor is not defined on a boundary element. We probably do not have a test for that.

Reply:
I get a failure with current moose with 4 procs and distributed mesh. Some day we'll have distributed mesh sweeps on modules

Reply:
> Note that `neighbor->is_remote()` does not mean we cannot use neighbor values. So I think skipping the ghost elements in this loop is possibly a better fix.

What are you talking about? You're going to try to evaluate a functor with a remote element? That in general will not work. The only things that could potentially work for are postprocessors/Functions, but it will certainly fail for variable functors

Reply:
We basically check neighbor of neighbors of local elements here. So if that is remote, it is fine still for computeResidual. I can revert my change and add a if-continue for checking whether the element is local. Do you think that is okay?

Reply:
I pushed what I think the code should be

Reply:
I pushed another commit, which fixes the `functor.error` test.